packages/server/engine-lib/rocketlib-python/lib/depends.py

```
@@ -874,10 +882,11 @@ def main():
             constraints_path = os.path.join(cache_dir, 'constraints.txt')
             if sys.argv[1] in ('install', 'sync'):
                 if os.path.exists(constraints_path) and os.path.getsize(constraints_path) > 0:
-                    uv_args.extend(['-c', constraints_path])
+                    uv_args.extend(['-c', './cache/constraints.txt'])
 
             # Run uv
-            result = subprocess.run(uv_args)
+            cmd = _build_cmd(uv_args)
+            result = subprocess.run(cmd, shell=True, cwd=exe_dir)
```

Ruff (0.15.6)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)